frontend/app/src/App.tsx

```diff
@@ -1508,19 +1506,17 @@ export class App extends PureComponent<Props, State> {
 
   showDeployButton = (): boolean => {
     return (
-      isTesting() ||
-      (SHOW_DEPLOY_BUTTON &&
-        showDevelopmentOptions(this.state.isOwner, this.state.toolbarMode) &&
-        !this.isInCloudEnvironment() &&
-        this.sessionInfo.isSet &&
-        !this.sessionInfo.isHello)
+      showDevelopmentOptions(this.state.isOwner, this.state.toolbarMode) &&
```

Comment:
?: Why remove the `isTesting` check?

Reply:
Hmm, I think my initial interpretation was that this is only something for testing it manually since it was deactivated behind the feature flag. But it is used with our e2e tests here:

https://github.com/streamlit/streamlit/blob/a436638889c18bd99ad4d21a350ba627afc3aaa3/e2e/specs/deploy_button.spec.js#L19

But it seems that we actually don't need this for testing since the e2e environment returns true for `showDevelopmentOptions`. The `isTesting` pattern based on query parameter is currently only used for this specific case. I'm wondering if we should go one step back here and think about if this is the best pattern going forward. In general, it might be good if the web app knows that it's running in e2e testing mode ... similar to how it knows that it's running in development mode:

https://github.com/streamlit/streamlit/blob/a436638889c18bd99ad4d21a350ba627afc3aaa3/frontend/lib/src/baseconsts.ts#L33

I will bring this up in our next standup.

Reply:
> The `isTesting` pattern based on query parameter is currently only used for this specific case.

Since this is the only usage, should perhaps also remove the `isTesting` function from `utils.ts` ?

Reply:
Yep, I will do that 👍